cql3/statements/select_statement.cc

```diff
@@ -152,13 +153,18 @@ select_statement::select_statement(schema_ptr schema,
     , _ordering_comparator(std::move(ordering_comparator))
     , _stats(stats)
     , _ks_sel(::is_system_keyspace(schema->ks_name()) ? ks_selector::SYSTEM : ks_selector::NONSYSTEM)
+    , _attrs(std::move(attrs))
 {
     _opts = _selection->get_query_options();
     _opts.set_if<query::partition_slice::option::bypass_cache>(_parameters->bypass_cache());
     _opts.set_if<query::partition_slice::option::distinct>(_parameters->is_distinct());
     _opts.set_if<query::partition_slice::option::reversed>(_is_reversed);
 }
 
+db::timeout_clock::duration select_statement::get_timeout(const query_options& options) const {
+    return _attrs->is_timeout_set() ? _attrs->get_timeout(options) : options.get_timeout_config().*get_timeout_config_selector();
+}
+
```

Comment:
Maybe this should be a free function to avoid the repetition. It can accept a timeout config selector as an extra parameter.

Reply:
I'd rather get rid of `get_timeout_config_selector` and just override the whole `get_timeout` routine during the prepare stage - it would probably result in cleaner code. Pointers-to-members are cool, but each time I see one, I spend 5 minutes on cppreference trying to recall what's their exact syntax. But that's an idea for a follow-up.

Reply:
Ok.